dvc/fs/utils.py

```diff
@@ -54,7 +54,11 @@ def transfer(
         try:
             return _link(from_fs, from_info, to_fs, to_info, hardlink=hardlink)
         except OSError as exc:
-            if exc.errno not in [errno.EXDEV, errno.ENOTSUP]:
+            if exc.errno not in [
+                errno.EXDEV,
+                errno.ENOTSUP,
+                errno.ENOSYS,  # https://github.com/iterative/dvc/issues/6962
```

Comment:
Exception message says `OSError [Errno 38]`, but the "function not implemented" is actually "ENOSYS".

Reply:
Also, I guess we need it in line 40 above, since this could happen for reflink and not happen for hardlink. Theoretically, at least.